crates/zksync/core/src/vm/runner.rs

```diff
@@ -264,8 +264,8 @@ where
     }
     let max_fee_per_gas = fix_l2_gas_price(ecx.env.tx.gas_price.to_u256());
 
-    let gas_limit = if paymaster_params.paymaster != Address::ZERO.to_h160() {
-        // If paymaster is set, use the proposed gas limit
+    // We check if the paymaster is set, if it is not set, we use the proposed gas limit
+    let gas_limit = if paymaster_params.paymaster.is_zero() {
```

Comment:
Please extract the check to use paymaster in a separate variable to avoid implicit logic of `is_zero()` here.